### PR TITLE
Inherit Ownable directly from GoldToken and removed Freezable from GoldToken.

### LINKED ISSUE #6662: GoldToken.sol should inherit directly from Ownable
### Expected Behavior

Should have Ownable directly

### Current Behavior

Has Ownable via Freezable (via UsingRegistry, that Freezable inherits).

### PR DESCRIPTION
**Pull request by deepakhb2**
### Description

GoldToken.sol should inherit directly from Ownable 
Expected Behavior
Should have Ownable directly

Current Behavior
Has Ownable via Freezable

### Other changes

NA

### Tested
Tested by running build command locally and tests have passed as part of pipeline.

### Related issues

- Fixes #6662

### Backwards compatibility

NA

### CODE REVIEW

File: packages/protocol/contracts/common/GoldToken.sol [hunk 1]
```diff
@@ -2,17 +2,17 @@ pragma solidity ^0.5.13;
...
 import "./CalledByVm.sol";
-import "./Freezable.sol";
 import "./Initializable.sol";
 import "./interfaces/ICeloToken.sol";
 import "../common/interfaces/ICeloVersionedContract.sol";
 
 contract GoldToken is
   Initializable,
   CalledByVm,
-  Freezable,
```
Comment (yorhodes) [inline packages/protocol/contracts/common/GoldToken.sol:15]: If we remove the `Freezable` interface also, we need to remove the `onlyWhenNotFrozen` modifiers as well
https://app.circleci.com/pipelines/github/celo-org/celo-monorepo/33230/workflows/e3014f01-e2fa-4e5f-b5d8-b84d7a1588cf/jobs/422147/parallel-runs/0/steps/0-114?invite=true

**Comment by yorhodes** [comment]
looks like you need to import `UsingRegistry.sol` as well

**Comment by deepakhb2** [comment]
@yorhodes Also, GoldToken is dependent on UsingRegistry by using setRegistry method. Looks like things are getting complicated. What is main purpose for inheriting Ownable directly?
importing will not solve as dependency is with modifier. I will have to inherit UsingRegistry and it is inheriting Ownable.

**Comment by yorhodes** [comment]
> What is main purpose for inheriting Ownable directly?

We want the inheritance to be explicit. We should inherit from both `Ownable` and `UsingRegistry` @deepakhb2

**Comment by deepakhb2** [comment]
@yorhodes Need your input on failed test protocol-test-release which is 'Required' to be passed. Please let me if any failed test is related to this change. Thank you.

**Comment by m-chrzan** [comment]
> @yorhodes Need your input on failed test protocol-test-release which is 'Required' to be passed. Please let me if any failed test is related to this change. Thank you.

The automatic contract version checker failed since GoldToken has been changed without changing its version number (in its `getVersionNumber` function). If you're interested, you can read more about Celo's smart contract versioning [here](https://docs.celo.org/community/release-process/smart-contracts#versioning).

Just setting the version to `(1, 1, 1, 1)` should fix the current issue in CI.

**Comment by deepakhb2** [comment]
@m-chrzan Thank you. I will check the documentation.

**Comment by deepakhb2** [comment]
Thank you @nambrot for the merge that fixed the tests.

**Comment by kevjue** [comment]
@deepakhb2  - Our Circle CI `certora-test` is having problems running when PRs from forked repos.  I created a [replicated PR](https://github.com/celo-org/celo-monorepo/pull/7493) that originates from our celo-monorepo, so that `certora-test` will run.

Thanks for making these changes!  We will figure out how we can get that test running for PRs like yours.

**Comment by deepakhb2** [comment]
Thank you @kevjue. I will use the this repo for my next issue until this issue is fixed.

**Comment by deepakhb2** [comment]
I think only core developers can push to this repo. I will continue working on forked repo.

**Pull request closed without merging**